inference/enterprise/workflows/core_steps/models/foundation/ocr.py

```diff
@@ -0,0 +1,156 @@
+
+from pydantic import ConfigDict, Field
+
+from inference.core.entities.requests.doctr import DoctrOCRInferenceRequest
+from inference.core.env import (
+    HOSTED_CORE_MODEL_URL,
+    LOCAL_INFERENCE_API_URL,
+    WORKFLOWS_REMOTE_API_TARGET,
+    WORKFLOWS_REMOTE_EXECUTION_MAX_STEP_BATCH_SIZE,
+    WORKFLOWS_REMOTE_EXECUTION_MAX_STEP_CONCURRENT_REQUESTS,
+)
+from inference.core.managers.base import ModelManager
+from inference.enterprise.workflows.core_steps.common.utils import (
+    attach_parent_info,
+    attach_prediction_type_info,
+    load_core_model,
+)
+from inference.enterprise.workflows.entities.base import OutputDefinition
+from inference.enterprise.workflows.entities.types import (
+    BATCH_OF_STRING_KIND,
+    PARENT_ID_KIND,
+    PREDICTION_TYPE_KIND,
+    FlowControl,
+    InferenceImageSelector,
+    OutputStepImageSelector,
+)
+from inference.enterprise.workflows.prototypes.block import (
+    WorkflowBlock,
+    WorkflowBlockManifest,
+)
+from inference_sdk import InferenceConfiguration, InferenceHTTPClient
+
+LONG_DESCRIPTION = """
+Retrieve the characters in an image using Optical Character Recognition (OCR).
+
+This block returns the text within an image.
+
+You may want to use this block in combination with a detections-based block (i.e. 
+ObjectDetectionBlock). An object detetcion model could isolate specific regions from an 
+image (i.e. a shipping container ID in a logistics use case) for further processing. 
+You can then use a CropBlock to crop the region of interest before running OCR.
+
+Using a detections model then croipping detections allows you to isolate your analysis 
+on particular regions of an image.
+"""
+
+
+class BlockManifest(WorkflowBlockManifest):
+    model_config = ConfigDict(
+        json_schema_extra={
+            "short_description": "Run Optical Character Recognition on a model.",
+            "long_description": LONG_DESCRIPTION,
+            "license": "Apache-2.0",
+            "block_type": "model",
+        }
+    )
+    type: Literal["OCRModel"]
+    name: str = Field(description="Unique name of step in workflows")
+    image: Union[InferenceImageSelector, OutputStepImageSelector] = Field(
+        description="Reference at image to be used as input for step processing",
+        examples=["$inputs.image", "$steps.cropping.crops"],
+    )
+
+
+class OCRModelBlock(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key"]
+
+    @classmethod
+    def get_input_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="result", kind=[BATCH_OF_STRING_KIND]),
```

Comment:
My understanding is that this would be handled within workflow- previous steps would have to crop and then this block would be executed on each individual crop

Reply:
Some OCR models/use-cases work that way but I think [DocTR](https://github.com/mindee/doctr) which we’re using can process a full document.

Though I wonder if we’re abusing it to do something it’s not good at. Maybe we should be using the underlying “text recognition” models its docs say it uses behind the scenes and ignoring its “text detection” functionality that it seems we don’t want/need.

Reply:
https://github.com/roboflow/inference/issues/419